ejb3/src/main/java/org/jboss/as/ejb3/subsystem/StrictMaxPoolResourceDefinition.java

```diff
@@ -172,6 +176,8 @@ public void registerAttributes(ManagementResourceRegistration resourceRegistrati
         for (AttributeDefinition attr : ads) {
             resourceRegistration.registerReadWriteAttribute(attr, null, osh);
         }
+
+        resourceRegistration.registerReadOnlyAttribute(DERIVED_SIZE, new StrictMaxPoolDerivedSizeReadHandler());
```

Comment:
The attribute can't be read successfully on a /profile=*/subsystem=ejb3 resource so it shouldn't be registered. To avoid that this can be guarded with:

```
if (resourceRegistration.getProcessType().isServer())
```

Reply:
@bstansberry I don't think that's necessary. Since https://issues.jboss.org/browse/WFCORE-1588 the org.jboss.as.controller.registry.ConcreteResourceRegistration#registerReadOnlyAttribute method will check if is correct to register the read-only attribute depending on the process type.

Reply:
Thanks @jmesnil,  you are right.